opal/util/show_help.c

```diff
@@ -229,46 +248,54 @@ static int open_file(const char *base, const char *topic)
          */
         for (i = 0; NULL != search_dirs[i]; i++) {
             filename = opal_os_path(false, search_dirs[i], base, NULL);
-            opal_show_help_yyin = fopen(filename, "r");
-            if (NULL == opal_show_help_yyin) {
-                opal_asprintf(&err_msg, "%s: %s", filename, strerror(errno));
-                base_len = strlen(base);
-                if (4 > base_len || 0 != strcmp(base + base_len - 4, ".txt")) {
-                    free(filename);
-                    opal_asprintf(&filename, "%s%s%s.txt", search_dirs[i], OPAL_PATH_SEP, base);
-                    opal_show_help_yyin = fopen(filename, "r");
+            if(filename) {
```

Comment:
Is there a case where `filename==NULL` here?

Reply:
Yes - in the case where `opal_show_help()` was called before `opal_show_help_init()` - `search_dirs` won't be populated yet.

Reply:
Probably not - I added it for extra caution. I can remove it if desired.

Reply:
I guess "defensive programming" is a valid reason.